tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -155,9 +155,16 @@ class SerializableModelLibraryGenerator {
         classBuilder.fields.add(_buildModelClassDBField(className));
 
         classBuilder.methods.add(_buildModelClassTableGetter());
+
+        classBuilder.implements
+            .add(refer('ConstraintSerialization', serverpodUrl(serverCode)));
       } else {
         classBuilder.extend =
             refer('SerializableEntity', serverpodUrl(serverCode));
+        if (serverCode) {
```

Comment:
Suggestion: Move the if check and the contents out to above line 147.

There is no need to nest this branching and repeat the code of implementing the interface.